metagraph/src/annotation/int_matrix/row_diff/int_row_diff.hpp

```diff
@@ -0,0 +1,343 @@
+#ifndef __INT_ROW_DIFF_HPP__
+#define __INT_ROW_DIFF_HPP__
+
+#include <algorithm>
+#include <iostream>
+#include <cassert>
+#include <string>
+#include <vector>
+
+#include "common/vectors/bit_vector_adaptive.hpp"
+#include "common/vector_map.hpp"
+#include "common/vector.hpp"
+#include "common/logger.hpp"
+#include "common/utils/template_utils.hpp"
+#include "graph/annotated_dbg.hpp"
+#include "graph/representation/succinct/boss.hpp"
+#include "graph/representation/succinct/dbg_succinct.hpp"
+#include "annotation/binary_matrix/row_diff/row_diff.hpp"
+#include "annotation/int_matrix/base/int_matrix.hpp"
+
+
+namespace mtg {
+namespace annot {
+namespace matrix {
+
+const size_t RD_PATH_RESERVE_SIZE = 2;
+
+
+/**
+ * Convert deltas to positive integer for enable compression:
+ *      0  -> X (not allowed, zero diffs must be skipped)
+ *      1  -> 0
+ *      -1 -> 1
+ *      2  -> 2
+ *      -2 -> 3
+ *      ...
+ */
+inline uint64_t encode_diff(int64_t x) {
+    assert(x);
+    return x > 0 ? (x - 1) * 2 : (-x) * 2 - 1;
```

Comment:
You can avoid the branch here by doing `(llabs(x) - 1) * 2 + signbit(x)`, but I doubt that implementing this differently will speed things up much